web-app/helpers/expiration-date.ts

```diff
@@ -0,0 +1,175 @@
+    diff_to_next_activity: number; // Number of days of inactivity if long (>=180 days) inactivity period is present for given account
+  },
+];
+
+/**
+ * issuedAtUnixNano is double casted in query because of Prisma template literal throwing 22P03 Error in DB
+ * https://github.com/prisma/prisma/issues/10424
+ * https://github.com/prisma/prisma/issues/5083
+ * Double casting : https://github.com/prisma/prisma/issues/4647#issuecomment-939555602
+ */
+// eslint-disable-next-line max-lines-per-function
+export function getRawQuery(accountName: string, issuedAtUnixNano: string) {
+  // Pulls from the public indexer. https://github.com/near/near-indexer-for-explorer#shared-public-access
+  /**
+   * This query uses Common Table Expressions(CTE) to execute two separate queries conditionally;
+   * the second query being executed if first query doesn't return any result.
+   * https://www.postgresql.org/docs/9.1/queries-with.html
+   * https://stackoverflow.com/a/68684814/10684149
+   */
+  /**
+   * ---- QUERY 1 ----
+   * First query checks if the account has a period where it hasn't been active for 180 days straight after the issue date
+   * IMPORTANT: When a certificate is issued to an account, it does not appear as a transaction on the account's transaction history. 
+   * Therefore, the period between the first mainnet activity and issue date needs to be checked as well whether it exceeds 180 days, or not.
+   * 
+   * This is done by: 
+   * 1. Calculating the difference between account's first activity after issue date and the issue date. 
+   * 2. Retrieving account activities after the issue date and checking differences between these dates in days.
+   *    
+   * Since it is important to detect the FIRST occurance of a >180-day period,
+   * only in case such period is present:
+   * -- 1. if difference between (first activity - issue date) > 180, moment is the issue date
+   * -- 2. if difference between (first activity - issue date) < 180, moment is the first occurence of >180-day period after the first activity
+   * 
+   * -- Example result: 
+   * [
+      { 
+        moment: '2019-08-03T00:00:00+00:00', // The beginning of the *first* such period (moment) + 180 days
+        diff_to_next_activity: 214 // Either (first activity - issue date) or any other period where >180-day inacitivity is present, in days
+      }
+    ]
+
+   * ---- QUERY 2 ----
+   * Second query is run if no 180-day-inactivity period is found and returns most recent activity date
+   * 
+   * -- Example result: 
+   * [
+      {
+        moment: '2022-04-07T16:25:59+00:00', // Most recent activity
+        diff_to_next_activity: null // null, since no >180-day period was found
+      }
+    ]
+   * 
+   * BOTH queries are set up to return the desired date in a column called 'moment'.
+   */
+
+  // *issue_date* <-----------Query - 1-----------> *last_activtiy*  <-----------Query - 2 -----------> *now (render date)*
```

Comment:
I'm hoping my comments do not add to confusion. Some clarification would be enough. 

I can't tell just by looking at the query if the following scenario is being covered? 

`ID-2022-01-01` <---- `Activity-2022-04-01` <--- `gt 6m` ---> `Activity-2022-10-15` <---  `render-2022-12-01`

The certification should be expired on 2022-10-01 (6 months have passed since the last activity on 2022-04-01)

Reply:
Nevermind, I think you have this test case to cover the scenario above.  https://github.com/NEAR-Edu/near-certification-tools/blob/feature/prisma-test-setup/web-app/test/__tests__/expiration-date.test.ts#L95